how-to/software/upgrade-your-release.md

```diff
@@ -1,41 +1,50 @@
+* **Fully update the current system:** The release upgrade process works best when the current system has all the latest updates installed. This is a standard **package upgrade**, **apt update** will refresh package lists (sync our package index with the upstream sources), and **apt upgrade** will install the latest versions of installed packages.
 
-  ```bash
-  sudo apt update
-  sudo apt upgrade
-  ```
+  1. Run these commands to ensure everything is up to date:
 
-* Check that there is enough free disk space for the upgrade. Upgrading a system will include downloading new packages, which is likely to be on the order of hundreds of new packages. Systems with additional software installed may therefore require a few gigabytes of free disk space.
+      ```bash
+      sudo apt update
+      sudo apt upgrade
+      ```
+  2. Confirm both commands complete successfully and no further updates are available.
 
-* The upgrade process takes time to complete. You should have dedicated time to participate in the upgrade process.
+  3. After applying all updates, **reboot your system** to ensure the latest kernel is running.
 
-* Third-party software repositories and personal package archives (PPAs) are disabled during the upgrade. However, any software installed from these repositories is not removed or downgraded. Software installed from these repositories is the most common cause of upgrade issues.
+* **Check that there is enough free space:** A release upgrade involves downloading hundreds of new packages, which can be several gigabytes. Make sure you have enough free disk space.
 
-* Backup all your data. Although upgrades are normally safe, there is always a chance that something could go wrong. It is extremely important that the data is safely copied to a backup location to allow restoration if there are any problems during the upgrade process.
+* **Dedicate time for the upgrade:** This process takes time. So we should dedicated period during which we can monitor the upgrade and respond to any prompts.
+
+* **Understand Third-party repositories:**  Third-party software repositories and Personal Package Archives (PPAs) are disabled during upgrade. While software installed from these sources will not be removed, it's the most common cause of upgrade issues. Be prepared to re-enable them or find updated versions after the upgrade.
+
+* **Backup all your data:**  Although upgrade are normally safe, there is always a chance that something could go wrong. It is extremely important that the data is safely copied to a backup location to allow restoration if any problems occur.
 
 ## Upgrade the system
 
+
 We recommend upgrading the system using the `do-release-upgrade` command on Server edition and cloud images. This command can handle system configuration changes that are sometimes needed between releases. To begin the process, run the following command:
 
+To start the process, run this command:
+
```

Comment:
You have to remove the "To begin the process, run the following command:" from the previous paragraph, or else these are two prompts.